pkgs/development/ocaml-modules/wayland/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, buildDunePackage
+, fetchurl
+, xmlm
+, lwt
+, logs
+, fmt
+, cstruct
+, cmdliner
+, alcotest-lwt
+}:
+
+buildDunePackage rec {
+  pname = "wayland";
+  version = "0.2";
+
+  useDune2 = true;
+
+  src = fetchurl {
```

Comment:
OOI why this vs fetchFromGitHub?

Reply:
That's a release tarball generated by `dune-release` and uploaded with the release manually (therefore there's `download` in the URL and not `archive`), so the hash should be stable and we can use `fetchurl`. This is pretty common for OCaml dependencies as they often preprocess the source for a release.

Reply:
Yeah, I saw it was a release artifact, I was just kinda... weirded out by the release source being different from the source source. I guess autotools does weird stuff too, but blergh.

Reply:
Iirc they often have `%%VERSION%%` strings everywhere which get substituted by the release tooling and similar stuff, so yea kinda weird.

Also opam uses these tarballs and I always copy the hash from there. That opam is using those archives is kind of good as it doesn't support post unpack hashing afaik.